cmd/subnet-utils.go

```diff
@@ -0,0 +1,385 @@
+		sh, pe := client.HelpConfigKV(globalContext, "subnet", "license", false)
+		fatalIf(probe.NewError(pe), "Unable to get config keys for subnet")
+
+		buf, e := client.GetConfigKV(globalContext, "subnet")
+		fatalIf(probe.NewError(e), "Unable to get server subnet config")
+
+		tgt, e := madmin.ParseSubSysTarget(buf, sh)
+		fatalIf(probe.NewError(e), "Unable to parse sub-system target subnet")
+
+		lic := tgt.KVS.Get("license")
+		if len(lic) > 0 {
+			return lic
+		}
+	}
+
+	return mcConfig().Aliases[aliasedURL].License
+}
+
+func mcConfig() *configV10 {
+	loadMcConfig = loadMcConfigFactory()
+	config, err := loadMcConfig()
+	fatalIf(err.Trace(mustGetMcConfigPath()), "Unable to access configuration file.")
+	return config
+}
+
+func minioConfigSupportsLicense(client *madmin.AdminClient) bool {
+	help, e := client.HelpConfigKV(globalContext, "", "", false)
+	fatalIf(probe.NewError(e), "Unable to get minio config keys")
+
+	for _, h := range help.KeysHelp {
+		if h.Key == "subnet" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func setSubnetLicenseConfig(aliasedURL string, lic string) {
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	if minioConfigSupportsLicense(client) {
+		configStr := "subnet license=" + lic
+		_, e := client.SetConfigKV(globalContext, configStr)
+		fatalIf(probe.NewError(e), "Unable to set subnet license config on minio")
+		return
+	}
+	mcCfg := mcConfig()
+	aliasCfg := mcCfg.Aliases[aliasedURL]
+	aliasCfg.License = lic
+	setAlias(aliasedURL, aliasCfg)
+}
+
+func getClusterRegInfo(admInfo madmin.InfoMessage, clusterName string) ClusterRegistrationInfo {
+	noOfPools := 1
+	noOfDrives := 0
+	for _, srvr := range admInfo.Servers {
+		if srvr.PoolNumber > noOfPools {
+			noOfPools = srvr.PoolNumber
+		}
+		noOfDrives += len(srvr.Disks)
+	}
+
+	return ClusterRegistrationInfo{
+		DeploymentID: admInfo.DeploymentID,
+		ClusterName:  clusterName,
+		UsedCapacity: admInfo.Usage.Size,
+		Info: ClusterInfo{
+			MinioVersion:    admInfo.Servers[0].Version,
+			NoOfServerPools: noOfPools,
+			NoOfServers:     len(admInfo.Servers),
+			NoOfDrives:      noOfDrives,
+			NoOfBuckets:     admInfo.Buckets.Count,
+			NoOfObjects:     admInfo.Objects.Count,
+		},
+	}
+}
+
+func generateRegToken(clusterRegInfo ClusterRegistrationInfo) (string, error) {
+	token, e := json.Marshal(clusterRegInfo)
+	if e != nil {
+		return "", e
+	}
+
+	return base64.StdEncoding.EncodeToString(token), nil
+}
+
+func subnetLogin() (string, error) {
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Subnet username: ")
+	username, _ := reader.ReadString('\n')
+	username = strings.Trim(username, "\n")
+
+	if len(username) == 0 {
+		return "", errors.New("Username cannot be empty. If you don't have one, please create one from here: " + minioSubscriptionURL)
+	}
+
+	fmt.Print("Password: ")
+	bytepw, _ := term.ReadPassword(int(syscall.Stdin))
```

Comment:
is this same as `terminal.ReadPassword`  and also this is better named as 

```
terminal.ReadPassword(int(os.Stdin.Fd()))
```

Reply:
ok, will change